### PR TITLE
Resolves #24 - Add push triggers option

### LINKED ISSUE #24: Allow for configuring what prompts a notification
We should provide the user a listing of events that could occur to which would prompt an event.  Example:
- [ ] New post
- [ ] Updated post
- [ ] New comment
  ....

### PR DESCRIPTION
**Pull request by darkwing**
Let me know what you think of this!

### CODE REVIEW

File: wp-web-push/wp-web-push-admin.php [hunk 1]
```diff
@@ -79,6 +114,17 @@ public function options() {
 </td>
 </tr>
 
+<tr>
+<th scope="row"><?php _e('Push Triggers', 'wpwebpush'); ?></th>
+<td>
+<fieldset>
+  <?php foreach($ALLOWED_TRIGGERS as $trigger): ?>
+  <label><input type="checkbox" name="webpush_triggers[]" value="<?php echo esc_attr__($trigger['key']); ?>" <?php echo in_array($trigger['key'], $triggers_option) ? 'checked' : '' ?> /> <?php _e($trigger['text'], 'wpwebpush'); ?></label><br />
```
Comment (marco-c) [inline wp-web-push/wp-web-push-admin.php]: We shouldn't localize the keys, since they're internal, right?

Comment (darkwing) [inline wp-web-push/wp-web-push-admin.php]: Oh, bleh, right.  Will update.

**Comment by marco-c** [comment]
Looks good!
We should also actually make use of the option you added (`webpush_triggers`) in wp-web-push-main.php and we should remove it in `on_uninstall` and create a default value in `on_activate`.

**Comment by darkwing** [comment]
Updated!

File: wp-web-push/wp-web-push-db.php [hunk 2]
```diff
@@ -56,6 +56,7 @@ public static function on_activate() {
     // Set default options.
     update_option('webpush_title', 'blog_title');
     update_option('webpush_min_visits', 3);
+    update_option('webpush_triggers', array());
```
Comment (darkwing) [inline wp-web-push/wp-web-push-db.php:59]: We _could_ default to `'new-post'`, we'll need to pull `$ALLOWED_TRIGGERS` into a file shared across the other plugin PHP files.

**Pull request merged**